framework/semaphore_pool.h

```diff
@@ -31,7 +31,7 @@ class SemaphorePool
 
 	SemaphorePool(const SemaphorePool &) = delete;
 
-	SemaphorePool(SemaphorePool &&other) = default;
```

Comment:
I think this is a typo. This should follow the design of the rest of the class and we should delete the move constructor.

The samples not build with `= delete`?
As a reference http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines C.21 backs this theory up.
`C.21: If you define or =delete any default operation, define or =delete them all`

Reply:
@jherico what do you think about this? Can you do the fix so we can merge this one too?